app/assets/javascripts/discourse/app/components/user-notification-schedule.js

```diff
@@ -0,0 +1,121 @@
+import Component from "@ember/component";
+import I18n from "I18n";
+import { action } from "@ember/object";
+import discourseComputed from "discourse-common/utils/decorators";
+
+export default Component.extend({
+  startTimeOptions: null,
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.set(
+      "startTimeOptions",
+      this.buildTimeOptions(0, {
+        includeNone: true,
+        showMidnight: false,
+      })
+    );
+  },
+
+  buildTimeOptions(startAt, opts = { includeNone: false, showMidnight: true }) {
+    let timeOptions = [];
+
+    if (opts.includeNone) {
+      timeOptions.push({
+        name: I18n.t("user.notification_schedule.none"),
+        value: -1,
+      });
+    }
+
+    for (let timeInMin = startAt; timeInMin <= 1440; timeInMin += 30) {
+      let hours = Math.floor(timeInMin / 60);
+      let minutes = timeInMin % 60;
+      let am = timeInMin < 720;
+      if (minutes === 0) {
+        minutes = "00";
+      }
+      if (hours === 0) {
+        hours = "12";
+      }
+      if (hours === 24) {
+        if (opts.showMidnight) {
+          timeOptions.push({
+            name: I18n.t("user.notification_schedule.midnight"),
+            value: 1440,
+          });
+        }
+        break;
+      }
+      if (!am && hours !== 12) {
+        hours = hours - 12;
+      }
+      timeOptions.push({
+        name: `${hours}:${minutes} ${am ? "AM" : "PM"}`,
+        value: timeInMin,
+      });
+    }
+    return timeOptions;
+  },
+
+  @action
+  startingTimeChangedForDay(dayIndex, val) {
+    this.model.set(
+      `user_notification_schedule.day_${dayIndex}_start_time`,
+      val
+    );
+    if (
+      val !== "-1" &&
+      this.model.user_notification_schedule[`day_${dayIndex}_end_time`] < val
+    ) {
+      this.model.set(
+        `user_notification_schedule.day_${dayIndex}_end_time`,
+        val + 30
```

Comment:
watchout will all this code, you compare val with string so I expect it's a string, you compare it with a value which I expect is an ineteger, and then you add integer to it.

You should know that:

```
"30" + 10 // "3010"
```

I would try to make the type of val clearer here from the beginning.